src/main/java/net/dv8tion/jda/api/requests/RestAction.java

```diff
@@ -764,6 +764,42 @@ default <O> RestAction<O> map(@Nonnull Function<? super T, ? extends O> map)
         return new MapRestAction<>(this, map);
     }
 
+    /**
+     * An intermediate operator that returns a modified RestAction.
+     *
+     * <p>This does not modify this instance but returns a new RestAction, which will consume
+     * the actions result using the given consumer on successful execution.
+     * The resulting action continues with the previous result.
+     *
+     * <p><b>Example</b><br>
+     * <pre>{@code
+     * public RestAction<String> retrieveMemberNickname(Guild guild, String userId) {
+     *     return guild.retrieveMemberById(userId)
+     *                 .map(Member::getNickname)
+     *                 .onSuccess(System.out::println);
+     * }
+     * }</pre>
+     *
+     * Prefer using {@link #queue(Consumer)} instead, if continuation of the action
+     * chain is not desired.
+     *
+     * @param  consumer
+     *         The consuming function to apply to the action result, failures are propagated
+     *         into the resulting action
+     *
+     * @return RestAction that consumes the action result
```

Comment:
```suggestion
     *
     * @throws IllegalArgumentException
     *         If the consumer is null
     *
     * @return RestAction that consumes the action result
```

Reply:
thats fair, note that the other methods in this class lack such an extra specification for their nonnull params